### PR TITLE
updating handle only when merging degrees

### PR DESCRIPTION
**Pull request by truhacevkir**

### CODE REVIEW

File: brage-import/src/test/java/no/sikt/nva/brage/migration/lambda/BrageTestRecord.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package no.sikt.nva.brage.migration.lambda;
+
+import no.sikt.nva.brage.migration.testutils.NvaBrageMigrationDataGenerator;
+import no.sikt.nva.brage.migration.testutils.NvaBrageMigrationDataGenerator.Builder;
+import no.unit.nva.model.Publication;
+
+public class BrageTestRecord {
+
```
Comment (truhacevkir) [inline brage-import/src/test/java/no/sikt/nva/brage/migration/lambda/BrageTestRecord.java:8]: Have no idea for better object name, but used to wrap/store existing publication in nva with brage record

File: brage-import/src/main/java/no/sikt/nva/brage/migration/merger/PreMergeValidator.java [hunk 2]
```diff
@@ -1,37 +1,27 @@
...
-        var handles = handleInAdditionalIdentifier(existingPublication);
-        if (isDegree(existingPublication) && !handles.isEmpty()) {
-            throw new DegreeMergingException(String.format(HANDLE_ALREADY_EXIST_ERROR_MESSAGE, handles));
-        }
+    public static boolean shouldNotMergePublications(Publication existingPublication) {
+        return isDegree(existingPublication) && hasBrageHandle(existingPublication);
     }
 
-    private static List<String> handleInAdditionalIdentifier(Publication existingPublication) {
+    private static boolean hasBrageHandle(Publication existingPublication) {
```
Comment (anetteOlli) [inline brage-import/src/main/java/no/sikt/nva/brage/migration/merger/PreMergeValidator.java:18]: Nice

File: brage-import/src/main/java/no/sikt/nva/brage/migration/merger/PreMergeValidator.java [hunk 3]
```diff
@@ -1,37 +1,27 @@
...
     private PreMergeValidator() {
 
     }
 
-    public static void validate(Publication existingPublication) {
-        var handles = handleInAdditionalIdentifier(existingPublication);
-        if (isDegree(existingPublication) && !handles.isEmpty()) {
-            throw new DegreeMergingException(String.format(HANDLE_ALREADY_EXIST_ERROR_MESSAGE, handles));
-        }
+    public static boolean shouldNotMergePublications(Publication existingPublication) {
```
Comment (anetteOlli) [inline brage-import/src/main/java/no/sikt/nva/brage/migration/merger/PreMergeValidator.java:14]: This method should now be renamed to something else.

Comment (truhacevkir) [reply brage-import/src/main/java/no/sikt/nva/brage/migration/merger/PreMergeValidator.java:14]: What about shouldNotMergeMatadata? So we differentiate between "merging metadata" vs "injecting handle only"

**Review by brinxmat — APPROVED** [review]

**Review by anetteOlli — APPROVED** [review]

**Pull request merged**